invenio/ext/jinja2hacks.py

```diff
@@ -106,18 +109,21 @@ def setup_app(app):
     return app
 
 
-def utf8escape(s):
-    """UTF8-8-bit-string-friendly replacement for MarkupSafe escape function.
+def _utf8escape(app):
+    def utf8escape(s):
+        """UTF8-8-bit-string-friendly replacement for MarkupSafe escape function.
 
-    WARNING: Do not use this method. Use jinja2.escape() instead.
-    """
-    if isinstance(s, str):
-        warnings.warn("Convert string '{0}' in template to unicode.".format(s),
-                      RuntimeWarning, stacklevel=3)
-        return jinja2_escape(s.decode('utf8'))
-    return jinja2_escape(s)
-# Ensure function name is identical to replaced function.
-utf8escape.__name__ = jinja2_escape.__name__
+        WARNING: Do not use this method. Use jinja2.escape() instead.
+        """
+        if isinstance(s, str):
+            if app.debug:
+                warnings.warn("Convert string '{0}' in template to unicode.".format(s),
+                              RuntimeWarning, stacklevel=3)
```

Comment:
Aight. I can set it to `DeprecationWarning` and see. Won't it still be raised by default and generate overhead unless we set some ignore deprecation warning flag somewhere?